api/src/pcapi/routes/native/v1/offers.py

```diff
@@ -25,11 +28,44 @@
 
 
 # It will break the WebApp v2 proxy in case of endpoint modification. Read https://github.com/pass-culture/pass-culture-app-native/pull/2808/files#r844891000
+# TODO (ASK, 2023-02-22): should be removed when no more app version use it
 @blueprint.native_v1.route("/offer/<int:offer_id>", methods=["GET"])
+@spectree_serialize(response_model=serializers.OfferLegacyResponse, api=blueprint.api, on_error_statuses=[404])
+def get_offer_legacy(offer_id: str) -> serializers.OfferLegacyResponse:
```

Comment:
(question) Pourquoi a-t-on besoin de créer une nouvelle route ? J'ai l'impression que la différence dans le schéma d'entrée/sortie est rétro-compatible, puisqu'on ne fait qu'ajouter un champ optionnel dans la réponse.

(Note : j'ai tenté de suivre [le lien](https://github.com/pass-culture/pass-culture-app-native/pull/2808/files#r844891000) indiqué dans le commentaire au-dessus de la fonction, mais l'ancre indiquée ("#r844891000") n'est pas trouvée. Donc je ne sais pas pourquoi on ne peut pas changer cet endpoint, ni si l'ajout d'un champ optionnel casserait quoi que ce soit dans le proxy. @kopax : est-ce que tu aurais un pointeur vers l'explication sur une pull request ou ailleurs ? Peut-être qu'on devrait rapatrier cette explication dans le code (ici et pour la route `/venue/<int:venue_id>` qui a le même commentaire) ?)

Reply:
on a créé une nouvelle route pour éviter d'avoir à forcer une nise à jour coté app native.

comme ça, les anciennes versions de l'app continuent de contacter l'ancienne route et ne risquent donc pas d'avoir de soucis à cause du nouveau champ renvoyé.

Reply:
... mais pourquoi les anciennes versions auraient-elles un soucis à cause du nouveau champ s'il est optionnel ?

Reply:
c'est du front et je suis pas compétent pour répondre, mais c'est arrivé déjà 2 fois depuis le début de l'année que je doive revert du code back en prod pour corriger une incompatibilité, alors je préfère blinder cette fois :)

pour plus de détail sur le soucis potentiel, j'invoque @dbenfouzari-pass @YassinL et @cedricls1 :)

Reply:
Ouais, ici tu n'introduis pas de breaking change pour le front
Petit article syympa pour résumer: https://iam-carrot.medium.com/handling-breaking-changes-in-an-api-81783cd29d89

Pas de breaking change : 

- Les trucs nouveaux. Les anciennes versions font comme si c'est pas là, les nouvelles les utilisent
  - nouvelle route
  - nouveau champ dans une réponse d'une route
- Les updates qui ne changent pas la structure 
  - Changer le calcul de la valeur d'un champ sans changer son nom ni son type

Breaking changes : 

- Les modifs
  - renommer un champ d'une route
  - renommer une route

- Les suppressions (plutot obvious :smile: )
  - supprimer une route 
  - supprimer un champ

Reply:
Effectivement si le paramètre est optionnel ça ne devrait rien casser.
On a été potentiellement trop refroidi par nos problèmes récents